File: src/main/java/tech/zolhungaj/amqapi/servercommands/gameroom/game/QuizPlayer.java
package tech.zolhungaj.amqapi.servercommands.gameroom.game;

import com.squareup.moshi.Json;
import tech.zolhungaj.amqapi.servercommands.objects.PlayerAvatar;

public record QuizPlayer(
		@Json(name = "name")
		String displayName,
		@Json(name = "level")
		int level,
		@Json(name = "gamePlayerId")
		int gamePlayerId,
		@Json(name = "host")
		boolean isHost,
		@Json(name = "avatarInfo")
		PlayerAvatar avatarInfo,
		@Json(name = "inGame")
		boolean inGame,
		@Json(name = "positionSlot")
		int positionSlot,
		@Json(name = "teamPlayer") Object teamPlayer,
		@Json(name = "teamNumber") Object teamNumber,
		@Json(name = "hidden") Object hidden,
		@Json(name = "slot") Object groupSlot,
		Object maxHp,
		Object hp,
		Object shield,
		Object abilityInfo,
		Object buffs,
		Object genreInfo,
		@Json(name = "playerName")
		Object playerNameForNexus, //???????
		@Json(name = "pose")
		int pose,
		@Json(name = "score") int score,
		@Json(name = "teamCaptain")
		Object teamCaptain,
		@Json(name = "multiChoiceActive")
		boolean isMultiChoiceActive,
		@Json(name = "position") Integer position
) {

}

File: src/main/java/tech/zolhungaj/amqapi/servercommands/objects/QuizPlayer.kt
package tech.zolhungaj.amqapi.servercommands.objects

import com.squareup.moshi.Json

@JvmRecord
data class QuizPlayer(
    @Json(name = "name") val playerName: String,
    @Json(name = "level") val level: Int,
    @Json(name = "gamePlayerId") val gamePlayerId: Int,
    @Json(name = "host") val isHost: Boolean,
    @Json(name = "avatarInfo") val avatarInfo: PlayerAvatar,
    @Json(name = "inGame") val inGame: Boolean,
    @Json(name = "teamPlayer") val teamPlayer: Boolean?,
    @Json(name = "teamNumber") val teamNumber: Int?,
    @Json(name = "pose") val pose: AvatarPose,
    @Json(name = "score") val score: Int,
    @Json(name = "teamCaptain") val teamCaptain: Boolean?,
    @Json(name = "multiChoiceActive") val hasMultiChoiceActive: Boolean,
    @Json(name = "position") val position: Int,
    @Json(name = "positionSlot") val positionSlot: Int?
) 